App crashes on boot: IllegalThreadStateException in PingForegroundService.onStartCommand (workingThread started more than once)

## Summary

When **Start on boot** is enabled, the app crashes immediately after the device boots with `java.lang.IllegalThreadStateException`, and enters a crash/restart loop (Android shows "<app> keeps stopping"). Launching the app **manually from the launcher works fine** — the crash is specific to the boot / service (re)start path.

Reproduced on **v1.65.1** (stable) and **v1.65.2**, and the cause is present in current `master` as well, so it is not version‑specific.

## Root cause

`PingForegroundService` keeps a **single** `workingThread` created with `thread(start = false, ...)` and calls `workingThread.start()` inside `onStartCommand()`. But `onStartCommand()` can be invoked **more than once** for the same service instance (e.g. `START_STICKY` redelivery / multiple start requests on boot). A `java.lang.Thread` can only be started **once**, so the second `start()` throws `IllegalThreadStateException`.

Relevant code (`master`, `PingForegroundService.kt`):

```kotlin
private val workingThread = thread(start = false, priority = Thread.MIN_PRIORITY) {
    val interval = settings.intervalSeconds ?: return@thread
    while (!stopRequested) { ... }
}

override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
    ...
    startForeground(NotificationsService.NOTIFICATION_ID_PING_SERVICE, notification)
    workingThread.start()   // <-- throws on the 2nd invocation
    return super.onStartCommand(intent, flags, startId)
}
```

## Stack trace

```
FATAL EXCEPTION: main
Process: me.capcom.smsgateway
java.lang.RuntimeException: Unable to start service me.capcom.smsgateway.modules.ping.services.PingForegroundService@... with Intent { cmp=me.capcom.smsgateway/.modules.ping.services.PingForegroundService }: java.lang.IllegalThreadStateException
    at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4825)
    at android.app.ActivityThread.access$2200(ActivityThread.java:275)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2205)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:210)
    at android.os.Looper.loop(Looper.java:299)
    at android.app.ActivityThread.main(ActivityThread.java:8273)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:576)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1073)
Caused by: java.lang.IllegalThreadStateException
    at java.lang.Thread.start(Thread.java:1417)
    at me.capcom.smsgateway.modules.ping.services.PingForegroundService.onStartCommand(PingForegroundService.kt:84)
    at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4807)
    ... 9 more
```

(On boot the service is restarted in quick succession, so the FATAL fires repeatedly — same trace, new PID each time.)

## Steps to reproduce (100%)

1. Install the app (reproduced with v1.65.1 and v1.65.2).
2. Configure **Cloud server**, enable **Start on boot**, confirm the app is ONLINE.
3. Reboot the device (`adb reboot`, or power off/on).
4. Within ~30 s – 2 min after boot, run `adb logcat -b crash -d` → the FATAL above appears every time.
5. Force‑stop, then start the app from the launcher → it runs normally with no crash. → confirms the bug is specific to the boot / service‑restart path, not the app's core function.

## Environment

- App: **v1.65.1** (stable) and **v1.65.2** — both reproduce.
- Device: **Android 12**, MIUI 13 (Xiaomi). Google Play services up to date; all SMS/phone/FCM permissions granted; battery optimizations disabled / app exempted.
- Because the cause is in app code (present in `master`), it should reproduce on any device whenever `onStartCommand` runs more than once.

## Suggested fix

`onStartCommand` must be idempotent with respect to the worker thread, because it can be called multiple times. A reusable `Thread` field cannot be `start()`ed twice (and a terminated `Thread` cannot be restarted either). Options:

- Move the worker start into `onCreate()` (called once per service instance) instead of `onStartCommand()`; or
- Create a fresh worker on each start and guard against double‑start, e.g.:

```kotlin
override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
    startForeground(NotificationsService.NOTIFICATION_ID_PING_SERVICE, notification)
    if (workingThread?.isAlive != true) {
        workingThread = thread(priority = Thread.MIN_PRIORITY) { /* ping loop */ }
    }
    return START_STICKY
}
```

- Or replace the raw `Thread` with a `ScheduledExecutorService` / coroutine `Job` that is safe to (re)schedule idempotently.

Happy to provide more logs if useful. Thanks for the project!

--- a/app/src/main/java/me/capcom/smsgateway/modules/ping/services/PingForegroundService.kt
+++ b/app/src/main/java/me/capcom/smsgateway/modules/ping/services/PingForegroundService.kt
@@ -70,6 +70,7 @@ class PingForegroundService : Service() {
 
         wakeLock.acquire()
         wifiLock.acquire()
+        workingThread.start()
     }
 
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
@@ -80,8 +81,6 @@ class PingForegroundService : Service() {
         )
 
         startForeground(NotificationsService.NOTIFICATION_ID_PING_SERVICE, notification)
-
-        workingThread.start()
 
         return super.onStartCommand(intent, flags, startId)
     }

